salt/_utils/kubernetes_utils.py

```diff
@@ -0,0 +1,535 @@
+"""Utility methods for manipulation of Kubernetes objects in Python."""
+import datetime
+import enum
```

Comment:
Oopsie, forgot that, so used to `enum`... Didn't find much when [searching their repo](https://github.com/saltstack/salt/search?q=enum&unscoped_q=enum) though.

I'll do some python2-compatible variant of `enum` then (don't want to go down the road of installing/importing the [`enum34` backport](https://pypi.org/project/enum34/)).

Reply:
IIRC, the py2-way of doing enum is simply to use class attribute like that:

```
class ObjectScope(object):
    NAMESPACE = 'namespaced'
    CLUSTER = 'cluster'
```

You don't get something as fancy as py3-enum, but for the use we have here it's probably good enough.

Reply:
I did something slightly more fancy, for comfort:

```python
# Roughly equivalent to an Enum, for Python 2
class ObjectScope:
    NAMESPACE = 'namespaced'
    CLUSTER = 'cluster'

    def __init__(self, value):
        allowed_values = (ObjectScope.NAMESPACE, ObjectScope.CLUSTER)
        if value not in allowed_values:
            raise ValueError('Value must be one of {}.'.format(allowed_values))
        self.value = value

    def __eq__(self, other):
        if isinstance(other, ObjectScope):
            return self.value == other.value
        if isinstance(other, str):
            return self.value == other
        return NotImplemented
```

Reply:
I guess `allowed_values` don't need to be recreated at each `__init__` call and could use `basestring` (or whatever `six` provides` in the `isinstance` instead of `str`).

Otherwise LGTM :slightly_smiling_face: